### PR TITLE
[Relay] Use typings for Modern

### PR DESCRIPTION
**Pull request by alloy**
From https://github.com/DefinitelyTyped/DefinitelyTyped/pull/19533

Skip New Tests

### CODE REVIEW

File: src/lib/Components/Gene/Artworks.tsx [hunk 1]
```diff
@@ -1,27 +0,0 @@
...
-        artworks={[]}
-        queryState={this.props.queryState}
-        queryForPage={this.props.resolveQuery}
-        queryArtworksKeypath="gene.filtered_artworks.hits"
-      />
-    )
-  }
-}
-
-export default Artworks
```
Comment (alloy) [inline src/lib/Components/Gene/Artworks.tsx]: This file was unused.

File: src/lib/Components/ArtworkGrids/InfiniteScrollGrid.tsx [hunk 2]
```diff
@@ -49,37 +49,37 @@ interface ArtworksConnection {
...
-  itemMargin: number
+  itemMargin?: number
 
   /** The artist in question */
-  artist: any
+  artist?: any
 
   /** The gene in question */
-  gene: any
+  gene?: any
```
Comment (alloy) [inline src/lib/Components/ArtworkGrids/InfiniteScrollGrid.tsx]: This component knowing of both artist and gene is a smell, it should be more generic.

File: package.json [hunk 3]
```diff
@@ -86,6 +78,8 @@
     "@types/lodash": "^4.14.64",
     "@types/react": "^16.0.5",
     "@types/react-native": "^0.48.0",
+    "@types/react-relay": "/Users/eloy/Code/TypeScript/types-publisher/output/react-relay/types-react-relay-1.3.0.tgz",
+    "@types/relay-runtime": "/Users/eloy/Code/TypeScript/types-publisher/output/relay-runtime/types-relay-runtime-1.3.0.tgz",
```
Comment (alloy) [inline package.json]: :warning: Temporary

File: src/lib/Components/ArtworkGrids/InfiniteScrollGrid.tsx (hunk 2 above)
Comment (alloy) [reply src/lib/Components/ArtworkGrids/InfiniteScrollGrid.tsx]: There’s now also the ‘sales’ grid, which will suffer from the same. /cc @ashfurrow

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,8 +1,14 @@
...
 // These docs are aimed at developers, but are shown inside collector GMV's channel
 
+### Master
+
+-   Allow passing a set of Artworks and an index to load to Eigen when clicking on
+    a Artwork within a grid - orta
+
 ### 1.4.0-beta.9
 
+-   During Inbox pull-to-refresh, Conversations container force-fetches itself - sarah
```
Comment (alloy) [inline CHANGELOG.md]: I don’t know why these show up, as they’re already in master :?

File: src/lib/Components/ArtworkGrids/RelayConnections/SaleArtworksGrid.tsx [hunk 5]
```diff
@@ -1,13 +1,21 @@
...
 import Artwork from "../Artwork"
 // tslint:disable-next-line:no-unused-expression
 // This is so that TypeScript won’t remove the seemingly unused `Artwork` import. Relay depends on it to exist.
 Artwork
 
-const SaleArtworksGrid = createPaginationContainer(
-  InfiniteScrollArtworksGrid,
+// FIXME: This is really only done because the scrollgrid does not accept a `sale` prop and currently the way things are
+//        setup type-checking fails because users correctly do pass a `sale` prop. This should be made correct when work
+//        is continued on a RN version of a Sale.
```
Comment (alloy) [inline src/lib/Components/ArtworkGrids/RelayConnections/SaleArtworksGrid.tsx]: cc @ashfurrow

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  No CHANGELOG added.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/ArtworkGrids/RelayConnections/__tests__/ArtistForSaleArtworksGrid-tests.tsx`
- `src/lib/Components/Inbox/ActiveBids/__tests__/index-tests.tsx`
- `src/lib/Components/Inbox/Conversations/__tests__/MarkReadMessage-tests.ts`
- `src/lib/Components/Inbox/Conversations/__tests__/SendConversationMessage-tests.ts`
- `src/lib/Components/Inbox/Conversations/__tests__/index-tests.tsx`
- `src/lib/Scenes/Favorites/Components/Artworks/__tests__/index-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/lib/Components/ArtworkGrids/InfiniteScrollGrid.tsx [hunk 6]
```diff
@@ -48,40 +48,40 @@ interface ArtworksConnection {
   }>
 }
 
-interface Props extends ArtistRelayProps, GeneRelayProps {
+export interface Props extends ArtistRelayProps, GeneRelayProps {
```
Comment (alloy) [inline src/lib/Components/ArtworkGrids/InfiniteScrollGrid.tsx]: Currently only for the ‘sale’ artwork grid.

**Comment by alloy** [comment]
@damassi You probably need to rebase on top of this. Let me know as early as possible if you run into type issues that you don’t know how to solve yet so I can be of assistance!

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/ArtworkGrids/RelayConnections/__tests__/ArtistForSaleArtworksGrid-tests.tsx`
- `src/lib/Components/Inbox/ActiveBids/__tests__/index-tests.tsx`
- `src/lib/Components/Inbox/Conversations/__tests__/MarkReadMessage-tests.ts`
- `src/lib/Components/Inbox/Conversations/__tests__/SendConversationMessage-tests.ts`
- `src/lib/Components/Inbox/Conversations/__tests__/index-tests.tsx`
- `src/lib/Scenes/Favorites/Components/Artworks/__tests__/index-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/ArtworkGrids/RelayConnections/__tests__/ArtistForSaleArtworksGrid-tests.tsx`
- `src/lib/Components/Inbox/ActiveBids/__tests__/index-tests.tsx`
- `src/lib/Components/Inbox/Conversations/__tests__/MarkReadMessage-tests.ts`
- `src/lib/Components/Inbox/Conversations/__tests__/SendConversationMessage-tests.ts`
- `src/lib/Components/Inbox/Conversations/__tests__/index-tests.tsx`
- `src/lib/Scenes/Favorites/Components/Artworks/__tests__/index-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by damassi** [comment]
@alloy - Will do, one sec...

File: src/lib/Components/ArtworkGrids/InfiniteScrollGrid.tsx (hunk 2 above)
Comment (damassi) [reply src/lib/Components/ArtworkGrids/InfiniteScrollGrid.tsx]: I was thinking about this the other day, and it might be time to backport the backport back to Emission: https://github.com/artsy/force/blob/master/desktop/components/react/masonry_grid/MasonryGrid.js

When I was working on new Force responsive views I needed a grid that allowed me to pass in any kind of artwork and made a few modifications: https://github.com/artsy/force/blob/master/desktop/apps/auction/components/artwork_browser/main/ArtworkDisplay.js#L84

File: src/lib/Components/Inbox/ActiveBids/index.tsx [hunk 7]
```diff
@@ -90,7 +90,7 @@ export default createRefetchContainer(
       }
     }
   `
-)
+) as React.ComponentClass<Props>
```
Comment (damassi) [inline src/lib/Components/Inbox/ActiveBids/index.tsx]: Why is this necessary here? Isn't this info encoded in the `class ActiveBids extends React.Component<Props, State>` signature?

File: src/lib/Components/Inbox/Conversations/Messages.tsx [hunk 8]
```diff
@@ -26,7 +26,7 @@ interface State {
 const LoadingIndicator = styled.ActivityIndicator`margin-top: 40px;`
 
 export class Messages extends React.Component<Props, State> {
-  constructor(props) {
+  constructor(props: Props) {
```
Comment (damassi) [inline src/lib/Components/Inbox/Conversations/Messages.tsx]: Since we're not using `props` in the constructor could move `state` out and eliminate this entire block

File: src/lib/Scenes/Home/Components/Sales/index.tsx [hunk 9]
```diff
@@ -116,7 +116,7 @@ export default createPaginationContainer(
     `,
   },
   {
-    getConnectionFromProps: ({ viewer }) => viewer && viewer.sale_artworks,
+    getConnectionFromProps: ({ viewer }) => viewer && (viewer.sale_artworks as ConnectionData),
```
Comment (damassi) [inline src/lib/Scenes/Home/Components/Sales/index.tsx]: In my branch this block is no longer in this file but I've made sure to update on my side

File: src/lib/Components/ArtworkGrids/InfiniteScrollGrid.tsx (hunk 2 above)
Comment (alloy) [reply src/lib/Components/ArtworkGrids/InfiniteScrollGrid.tsx]: Something like that yeah, let’s pair on that. The thought I had was that all sorts of types can have artwork connections and we could just have them implement an `ArtworkInterface` on their edge so that we can pass in any connection that implements that interface.

File: src/lib/Components/Inbox/ActiveBids/index.tsx (hunk 7 above)
Comment (alloy) [reply src/lib/Components/Inbox/ActiveBids/index.tsx]: Correct. I need to think a bit on how to fix this, but this is imo a bug with the Relay typings. It specifies that `create*Container` returns a union of `React.ComponentClass` and `React.StatelessComponent`, the latter has no `ref` prop which the user of this component needs.

File: src/lib/Components/Inbox/Conversations/Messages.tsx (hunk 8 above)
Comment (alloy) [reply src/lib/Components/Inbox/Conversations/Messages.tsx]: Good point! Also a bit above.

File: src/lib/Scenes/Home/Components/Sales/index.tsx (hunk 9 above)
Comment (alloy) [reply src/lib/Scenes/Home/Components/Sales/index.tsx]: 👍

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/ArtworkGrids/RelayConnections/__tests__/ArtistForSaleArtworksGrid-tests.tsx`
- `src/lib/Components/Inbox/ActiveBids/__tests__/index-tests.tsx`
- `src/lib/Components/Inbox/Conversations/__tests__/MarkReadMessage-tests.ts`
- `src/lib/Components/Inbox/Conversations/__tests__/SendConversationMessage-tests.ts`
- `src/lib/Components/Inbox/Conversations/__tests__/index-tests.tsx`
- `src/lib/Scenes/Favorites/Components/Artworks/__tests__/index-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by alloy** [comment]
Alright, going for it.

**Comment by damassi** [comment]
Rebasin'

**Comment by sarahscott** [comment]
👏
